packages/app-store/closecomothercalendar/pages/setup/index.tsx

```diff
@@ -0,0 +1,157 @@
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useRouter } from "next/router";
+import { useState, useEffect } from "react";
+import { Check, X } from "react-feather";
+import { useForm, Controller } from "react-hook-form";
+import { Toaster } from "react-hot-toast";
+import z from "zod";
+
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import showToast from "@calcom/lib/notification";
+import Button from "@calcom/ui/v2/Button";
+import { Form, TextField } from "@calcom/ui/v2/form/fields";
+
+const formSchema = z.object({
+  api_key: z.string(),
+});
+
+export default function CloseComSetup() {
+  const { t } = useLocale();
+  const router = useRouter();
+  const [testPassed, setTestPassed] = useState<boolean | undefined>(undefined);
+  const [testLoading, setTestLoading] = useState<boolean>(false);
+
+  const form = useForm<{
+    api_key: string;
+  }>({
+    resolver: zodResolver(formSchema),
+  });
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      if (testPassed === false) {
+        setTestPassed(undefined);
+      }
+    }, 3000);
+    return () => clearTimeout(timer);
+  }, [testPassed]);
+
+  return (
+    <div className="flex h-screen bg-gray-200">
+      <div className="m-auto rounded bg-white p-5 md:w-[520px] md:p-10">
+        <div className="flex flex-col space-y-5 md:flex-row md:space-y-0 md:space-x-5">
+          <div>
+            {/* eslint-disable @next/next/no-img-element */}
+            <img
+              src="/api/app-store/closecomothercalendar/icon.svg"
+              alt="Apple Calendar"
+              className="h-12 w-12 max-w-2xl"
+            />
+          </div>
+          <div>
+            <h1 className="text-gray-600">{t("provide_api_key")}</h1>
+
+            <div className="mt-1 text-sm">
+              {t("generate_api_key_description")}{" "}
+              <a
+                className="text-indigo-400"
+                href="https://app.close.com/settings/api/"
+                target="_blank"
+                rel="noopener noreferrer">
+                Close.com
+              </a>
+              . {t("it_stored_encrypted")}
+            </div>
+            <div className="my-2 mt-3">
+              <Form
+                form={form}
+                handleSubmit={async (values) => {
+                  const res = await fetch("/api/integrations/closecomothercalendar/add", {
+                    method: "POST",
+                    body: JSON.stringify(values),
+                    headers: {
+                      "Content-Type": "application/json",
+                    },
+                  });
+                  const json = await res.json();
+
+                  if (res.ok) {
+                    router.push(json.url);
+                  } else {
+                    showToast(json.message, "error");
+                  }
```

Comment:
Can we adopt a similar style as [routing forms](https://github.com/calcom/cal.com/blob/feat/closecom-app/packages/app-store/ee/routing_forms/trpc-router.ts#L0-L1) here?

Reply:
@zomars I guess we could, but to be honest, it feels a bit too much for what needs to be done compared to routing forms. Are we looking to standardize that kind of handling? If so, makes sense to refactor it.

Reply:
We're moving away from api handlers in web app but we'll let it slide for now.